### PR TITLE
Story/um 767 get allocator docs

### PR DESCRIPTION
**Pull request by kab163**
- [ ] Readability of documentation fulfills the purpose of JIRA ticket?
- [x] Is it sufficient to have examples I have currently in PR or should there be a different kind of example?
- [x] Modifying example to use SizeLimiter strategy instead (better use case)

### CODE REVIEW

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
Resulting ReadTheDocs pages:
https://umpire.readthedocs.io/en/story-um-767-getallocator-docs/tutorial/dynamic_pool.html
https://umpire.readthedocs.io/en/story-um-767-getallocator-docs/tutorial/allocators.html

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Review by mcfadden8 — APPROVED** [review]
Looks great to me.  Thanks!

File: docs/sphinx/tutorial/allocators.rst [hunk 1]
```diff
@@ -17,7 +17,16 @@ All :class:`umpire::Allocator` s are created and managed by Umpire's
...
 
-Once you have an :class:`umpire::Allocator` you can use it to allocate and deallocate memory:
+You can also use an existing allocator to build an additional allocator off of it:
+
+.. literalinclude:: ../../../examples/tutorial/tut_allocator.cpp
+   :start-after: _sphinx_tag_tut_getAllocator_start
+   :end-before: _sphinx_tag_tut_getAllocator_end
+   :language: C++
+
+This "add-on" allocator will also be built with the same memory resource. More information
```
Comment (davidbeckingsale) [inline docs/sphinx/tutorial/allocators.rst:27]:
```suggestion
This new allocator will also be built with the same memory resource. More information
```

**Comment by kab163** [comment]
LGTM for the win

**Review by davidbeckingsale — CHANGES_REQUESTED** [review]
One suggestion. I also think that we could say `build a new allocator from` instead of saying `off of`. I think the former is clearer.

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
LGTM

**Review by davidbeckingsale — APPROVED** [review]

**Pull request merged**